roles/sft-server/tasks/install.yml

```diff
@@ -6,26 +6,14 @@
   when:
     - sft_artifact_file_path is defined
 
-# - name: downloading SFT artifact by version
-#   # get_url
-#   debug:
-#     msg: "TODO"
-#   when:
-#     - sft_version is defined
-#     - sft_artifact_file_path is not defined
-#     - sft_artifact_file_url is not defined
-#     - sft_artifact_checksum is not defined
-
-# - name: downloading SFT artifact by URL
-#   # get_url
-#   debug:
-#     msg: "TODO"
-#   when:
-#     - sft_artifact_file_url is defined
-#     - sft_artifact_checksum is defined
-#     - sft_artifact_file_path is not defined
-#     - sft_version is not defined
-
+- name: download SFT artifact
+  get_url:
+    url: "{{ sft_artifact_file_url }}"
+    dest: "{{ sft_artifact_target_path }}"
+    # FUTUREWORK: check sha256sum
+    # sha256sum: "{{ sft_artifact_checksum | default(omit) }}"
+  when:
+    - sft_artifact_file_url is defined
```

Comment:
This seems to be the implementation only of L19-27. What about L9-17? What was your motivation to change the interface definition reflected by these 3 tasks?

Reply:
> This seems to be the implementation only of L19-27. What about L9-17?

I think we can always pass in the URL, it is anyways not possible to download by version. Even if it were I think we can just derive a URL from the version. But let's do that if and when we need to it.

> What was your motivation to change the interface definition reflected by these 3 tasks?

I am not sure what you mean, can you please clarify?

Reply:
> I think we can always pass in the URL, it is anyways not possible to download by version. Even if it were I think we can just derive a URL from the version. But let's do that if and when we need to it.

Since @jschaul started to put it on s3, it would surely be possible to download by version (very similar to [this](https://github.com/wireapp/ansible-sft/pull/8/files#diff-7d4854bd1fef85a011d5542a9e50ec09R17-R19) approach in combination with variables like [these](https://github.com/wireapp/ansible-sft/pull/8/files#diff-00ee6cc36478d9ce7f01fb88808ddb9fR6-R16) and tasks like [those](https://github.com/wireapp/ansible-sft/pull/8/files#diff-f19b22dcf3dba136d1557b191cd0c0cfR1-R31)).

>> What was your motivation to change the interface definition reflected by these 3 tasks?
> I am not sure what you mean, can you please clarify?

I guess, I'm just trying to understand your reasoning behind omitting the checksum part and not fully implementing the [downloading SFT artifact by URL](https://github.com/wireapp/ansible-sft/pull/11/files#diff-04b2c5dcb8908a77ee73c64a7c2d0f78L19-L27) task.

Reply:
TBH, I am not sure sure why Joé left the checksum part as FUTUREWORK. But I wouldn't mind if that is done along with when we actually start publishing the artefacts.
As for the other ways of downloading the artefact, I think we don't really need to provide more ways of this as I don't see us using multiple ways of downloading artefacts right now. And if in future we need to have multiple ways of doing so, we should implement this in future.

Reply:
FUTUREWORK checksum: We didn't agree yet on how we publish the checksums. That needs to happen first. I don't know what you had in mind with e.g. a `version install`, but it didn't appear implementable at all.

> This seems to be the implementation only of L19-27. What about L9-17? What was your motivation to change the interface definition reflected by these 3 tasks?

The 'interface definition' is a comment so far, so there is an interface without an implementation, and the interface is based on false assumptions (namely that there need to be three different ways to install the SFT artifact and use this role). Why three ways? We don't need three ways. I left the 'from local file' path in there in order not to break other code in cailleach; but I'd like to remove that entirely eventually.